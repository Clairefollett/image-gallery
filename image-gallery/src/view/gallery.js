import React from 'react';

function Gallery(props) {
    const galleryObj = props.array.map((schnoodle, index) => {
        return (
            <li key={index}>
                <p>Title: {schnoodle.title}</p>
                <img className='gallery-img' src={schnoodle.url} alt='' />
                <p>description: {schnoodle.description}</p> 
            </li>    
        )
    });
    return (
        <ul>{galleryObj}</ul>
    )
}

export {
    Gallery
};