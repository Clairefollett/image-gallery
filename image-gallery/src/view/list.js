import React from 'react';

function List(props) {
    const listObj = props.array.map((schnoodle, index) => {
        return (
          <li key={index}>
            <p>title: {schnoodle.title}</p>
            <p>description: {schnoodle.description}</p>
            <p>url: <a href={schnoodle.url}>{schnoodle.url}</a></p>
          </li>
        );
    });
    return (
      <ul>{listObj}</ul>
    );
}

List.propTypes = {
    array: React.PropTypes.array
};

export {
    List
};