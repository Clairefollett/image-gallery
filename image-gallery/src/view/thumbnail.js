import React from 'react';

export default function Thumbnail(props) {
    const thumbnail = props.array.map((schnoodle, index) => {
        return (
          <li key={index}>
            <p>title: {schnoodle.title}</p>
            <img className='thumbnail' src={schnoodle.url} alt='' height='100' width='100' />
          </li>
        );
    });
    return (
      <ul>{thumbnail}</ul>
    );
}

Thumbnail.propTypes = {
    array: React.PropTypes.array
};
