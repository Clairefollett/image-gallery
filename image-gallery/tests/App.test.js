/* eslint-disable */

import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../src/app';
import { List } from '../src/view/list';
import { Gallery } from '../src/view/gallery';
import { Thumbnail } from '../src/view/thumbnail'

describe('my buttons click handlers', () => {
    it('runs my test!', () => {
        const rendered = shallow(<App />);
        expect(toJson(rendered)).toMatchSnapshot();
    });
});

//none of my other tests will work