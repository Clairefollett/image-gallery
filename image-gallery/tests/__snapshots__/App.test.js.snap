exports[`my buttons click handlers runs my test! 1`] = `
<div>
  <ImageView
    onClick={[Function]}
    view={null} />
</div>
`;
